test/llvm-intrinsics/fp_to_arbitrary_size_int_intrinsic.ll

```diff
@@ -0,0 +1,57 @@
+;; Ensure @llvm.fptosi.sat.* and @llvm.fptoui.sat.* intrinsics are translated
+
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_INTEL_arbitrary_precision_integers -spirv-text -o - | FileCheck %s --check-prefix=CHECK-SPIRV
+; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_INTEL_arbitrary_precision_integers -o %t.spv
+; RUN: spirv-val %t.spv
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
+
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-n8:16:32:64"
```

Comment:
Data layout doesn't match the input LLVM IR ;-)

Reply:
I looked at the original testcase. Data layout is exactly the same as it is here. Can you please let me know how it does not match the input LLVM IR? Thanks

Reply:
> n8:16:32:64

There is no "2" and "4" which would stand for i2 and i4

Reply:
Or if not hacks, may be just aggressive inst combine should respect data layout?

Reply:
It seems the entries in datalayout are expected to be more of a guide that strict rules. for example, It is not documented anywhere that i2 is illegal if it is not listed in the target datalayout, thought it does sound counter-intuitive. I think we might have to assume that this LLVM IR is legal and proceed with the change here.

Thanks

Reply:
> It seems the entries in datalayout are expected to be more of a guide that strict rules. for example, It is not documented anywhere that i2 is illegal if it is not listed in the target datalayout, thought it does sound counter-intuitive. I think we might have to assume that this LLVM IR is legal and proceed with the change here.
> 
> Thanks

I agree. The fact the `datalayout` field does not explicitly "mention" `i2` doesn't prevent the LLVM optimizer to use i2 as an intermediat type during optimization IMO. The LLVM language spec simply states that the " data layout string that specifies how data is to be laid out in memory". There isn't anything that says using i2 is illegal.